### PR TITLE
Feat: survey create form validation logic, test code and refactoring

### LINKED ISSUE #101: [P0] Input check logic on Survey form page for authors
Add logic for checking inputs on SurveyFormPage

### PR DESCRIPTION
**Pull request by sanghyun1128**
- Validation logic
> - check empty input
> - check question list or option list is empty
> - check start and end date is valid

- Test code
> - initial rendering test
> - add / delete question and option logic test
> - change question type test
> - validation logic test

- Survey create form
> - show alert modal when validation failed
> - show confirm modal before submit

Closes #101

### CODE REVIEW

File: web/src/components/Button/MediumRectangleButton.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+`;
+
+interface MediumRectangleButtonProps {
+  displayText: string;
+  type: 'button' | 'submit' | 'reset' | undefined;
+  handleClickButton: () => void;
+  theme: DefaultTheme;
+}
+
+export default function MediumRectangleButton({
```
Comment (seiwonpark) [inline web/src/components/Button/MediumRectangleButton.tsx]: Well `MediumRectangleButton` isn't intuitive. (ex. How large it is?, How much the "medium" is like?) Rather putting options like `size`, `shape` would be better.

File: web/src/components/Button/SmallCircleButton.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  border: none;
+  transition: 250ms background ease;
+  cursor: pointer;
+
+  &:hover {
+    opacity: 0.8;
+  }
+`;
+
+interface SmallCircleButtonProps {
```
Comment (seiwonpark) [inline web/src/components/Button/SmallCircleButton.tsx]: Same above. `Circle` makes sense but **how small** it is? If this is universal component, then something general would fit

File: web/src/components/Modal/ConfirmModal.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  handleCancleClick: () => void;
+  handleConfimClick: () => void;
+}
+
+export default function ConfirmModal({
+  theme,
+  title,
+  level,
+  text,
+  handleCancleClick,
```
Comment (seiwonpark) [inline web/src/components/Modal/ConfirmModal.tsx]: Typo here. `handleCan'cel'Click`

File: web/src/components/Modal/ConfirmModal.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  handleConfimClick,
```
Comment (seiwonpark) [inline web/src/components/Modal/ConfirmModal.tsx]: Typo here. `handleConfi'r'mClick`

File: web/src/components/Modal/SurveyEnterModal.tsx [hunk 5]
```diff
@@ -167,7 +132,7 @@ interface ModalProps {
   theme: DefaultTheme;
 }
 
-export default function SurveyPageResultModal({ surveyItem, setEnterModalOpen, theme }: ModalProps) {
+export default function SurveyEnterModal({ surveyItem, setEnterModalOpen, theme }: ModalProps) {
```
Comment (seiwonpark) [inline web/src/components/Modal/SurveyEnterModal.tsx]: Could you modify this to `SurveyPreviewModal`?

File: web/src/components/SurveyForm/OptionList.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  text-decoration-style: dotted;
+  text-decoration-color: currentColor;
+  border: ${(props) => props.theme.border};
+  border-radius: ${(props) => props.theme.borderRadius};
+`;
+
+interface OptionListProps {
+  surveyData: SurveyCreateRequest;
+  questionId: number;
+  handleChangeOption: (event: React.ChangeEvent<HTMLInputElement>, questionId: number, optionId: number) => void;
```
Comment (seiwonpark) [inline web/src/components/SurveyForm/OptionList.tsx]: Please modify this to `handleOptionChange`

File: web/src/components/SurveyForm/OptionList.tsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+  handleClickButton: (event: React.MouseEvent<HTMLButtonElement>, questionId?: number, optionId?: number) => void;
```
Comment (seiwonpark) [inline web/src/components/SurveyForm/OptionList.tsx]: Please modify this to `handleButtonClick`

File: web/src/components/SurveyForm/QuestionForm.tsx [hunk 8]
```diff
@@ -85,29 +113,66 @@ export default function SubjectiveAnswerForm({
...
+        <EmptyObject />
+        <SmallCircleButton
+          displayText="+"
+          name="addQuestion"
+          handleClickButton={handleClickButton}
+          theme={theme}
+          questionId={questionId}
+        />
+        <SmallCircleButton
+          displayText="🗑️"
```
Comment (seiwonpark) [inline web/src/components/SurveyForm/QuestionForm.tsx]: lol

File: web/src/features/SurveyFormValidation.ts [hunk 9]
```diff
@@ -0,0 +1,113 @@
+import { QuestionType } from '../types/request/Question';
```
Comment (seiwonpark) [inline web/src/features/SurveyFormValidation.ts]: What is the purpose of this `features/` for? I see some util functions but what is the difference between this and  `utils/validate.ts` file? Less cohesion.

File: web/src/features/SurveyFormValidation.ts [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+  const dateDiff = end.getTime() - start.getTime();
+
+  if (dateDiff <= 0) return false;
+
+  if (dateDiff < 60000) return false;
+
+  return true;
+};
+
+export default function SurveyFormValidation(surveyData: SurveyCreateRequest): InputCheckResult {
```
Comment (seiwonpark) [inline web/src/features/SurveyFormValidation.ts]: Let me fix this.

File: web/src/routes/SurveyPages/SurveyFormPage.tsx [hunk 11]
```diff
@@ -52,109 +60,96 @@ const SurveyDataContainer = styled(ItemContainer)`
...
     setResultModalOpen(true);
   };
 
+  const validateSurveyData = () => {
+    const checkResult: InputCheckResult = SurveyFormValidation(surveyData);
+    setAlertNotification(checkResult.index);
+
+    switch (checkResult.message) {
+      case ValidationErrorMessage.NO_QUESTION:
+        setWarnText('하나 이상의 질문을 추가 해주세요');
```
Comment (seiwonpark) [inline web/src/routes/SurveyPages/SurveyFormPage.tsx]: Wrong expression. "추가 해주세요" -> "추가해 주세요"

**Review by seiwonpark — CHANGES_REQUESTED** [review]
Please update your code and re-request. Tests are all looking good!

File: web/src/features/SurveyFormValidation.ts (hunk 9 above)
Comment (sanghyun1128) [reply web/src/features/SurveyFormValidation.ts]: I just want to manage validation logic separately, not managed on the page. 
I forgot utils/validate.ts is exist... I'll put them it together

File: web/src/components/Button/MediumRectangleButton.tsx (hunk 1 above)
Comment (sanghyun1128) [reply web/src/components/Button/MediumRectangleButton.tsx]: I just changed to export styled component(without size)

File: web/src/utils/validate.ts [hunk 12]
```diff
@@ -43,3 +57,139 @@ export const validatePhoneNumber = (phoneNumber: string): boolean => {
...
+ * 1) Check all inputs are filled
+ * 2) Validate start date
+ * 3) Validate end date
+ * 4) Check survey contain at least 1 question
+ * 5) Check selective question contain at least 1 option
+ *
+ * @param {SurveyCreateRequest} surveyData Survey data
+ * @returns {InputCheckResult} Validate result message and error question index(-1 means SurveyDataForm)
+ */
+export const validateSurveyData = (surveyData: SurveyCreateRequest): InputCheckResult => {
```
Comment (seiwonpark) [inline web/src/utils/validate.ts:110]: Yeah thank you for separating this but still you also need to separate the logic explicitly. Let me handle this later.

**Review by seiwonpark — APPROVED** [review]
LGTM. Thanks except for the typos.

**Pull request merged**